pennylane/math/utils.py

```diff
@@ -169,7 +172,66 @@ def convert_like(tensor1, tensor2):
     return np.asarray(tensor1, like=interface)
 
 
-def get_interface(tensor):
+def get_interface(*values):
+    """Determines the correct framework to dispatch to given a tensor or a
+    sequence of tensor-like objects.
+
+    Args:
+        values (tensor_like, Sequence[tensor_like]): a sequence of tensor like objects
+
+    Returns:
+        str: the name of the interface
+
+    To determine the framework to dispatch to, the following rules
+    are applied:
+
+    * Tensors that are incompatible (such as Torch, TensorFlow and Jax tensors)
+      cannot both be present.
+
+    * Autograd tensors *may* be present alongside Torch, TensorFlow and Jax tensors,
+      but Torch, TensorFlow and Jax take precendence; the autograd arrays will
+      be treated as non-differentiable NumPy arrays. A warning will be raised
+      suggesting that vanilla NumPy be used instead.
+
+    * Vanilla NumPy arrays and SciPy sparse matrices can be used alongside other tensor objects;
+      they will always be treated as non-differentiable constants.
+    """
+
+    if len(values) == 1:
+        return _get_interface(values[0])
+
+    interfaces = {_get_interface(v) for v in values}
+
+    if len(interfaces - {"numpy", "scipy", "autograd"}) > 1:
+        # contains multiple non-autograd interfaces
+        raise ValueError("Tensors contain mixed types; cannot determine dispatch library")
+
+    non_numpy_scipy_interfaces = set(interfaces) - {"numpy", "scipy"}
+
+    if len(non_numpy_scipy_interfaces) > 1:
+        # contains autograd and another interface
+        warnings.warn(
+            f"Contains tensors of types {non_numpy_scipy_interfaces}; dispatch will prioritize "
+            "TensorFlow, PyTorch, and  Jax over autograd. Consider replacing Autograd with vanilla NumPy.",
+            UserWarning,
+        )
+
+    if "tensorflow" in interfaces:
+        return "tensorflow"
+
+    if "torch" in interfaces:
+        return "torch"
+
+    if "autograd" in interfaces:
+        return "autograd"
+
+    if "jax" in interfaces:
+        return "jax"
+
+    return "numpy"
+
+
+def _get_interface(tensor):

```

Comment:
How about describing it in the function name that the function acts on a single tensor-like object? E.g., `_get_interface_of_tensor` or so.

Right now `get_interface` and `_get_interface` have very similar names, but one is compatible with sequences too.

Reply:
Changed!